hedera-node/test-clients/README.md

```diff
@@ -1,249 +1,282 @@
+a simple mock, and the `HederaState` implementation uses in-memory data structures instead of a
+Merkle tree.
 
 ## Table of contents
-- [Structure of a `HapiSpec`](#structure-of-a-hapispec)
-- [`HederaNetwork` implementations](#hederanetwork-implementations)
-  * [`RemoteNetwork`](#remotenetwork)
-  * [`SubProcessNetwork`](#subprocessnetwork)
-  * [Embedded networks](#embedded-networks)
-- [Style guide](#style-guide)
-  * [The `@HapiTest` checklist](#the-hapitest-checklist)
-- [JUnit Jupiter integrations](#junit-jupiter-integrations)
-  * [Meta-annotations](#meta-annotations)
-  * [`SharedNetworkLauncherSessionListener`](#sharednetworklaunchersessionlistener)
-  * [`NetworkTargetingExtension`](#networktargetingextension)
-  * [`SpecNamingExtension`](#specnamingextension)
-  * [`SpecManagerExtension`](#specmanagerextension)
-  * [`SpecEntityExtension`](#specentityextension)
+
+-   [Structure of a `HapiSpec`](#structure-of-a-hapispec)
+-   [`HederaNetwork` implementations](#hederanetwork-implementations)
+    -   [`RemoteNetwork`](#remotenetwork)
+    -   [`SubProcessNetwork`](#subprocessnetwork)
+    -   [Embedded networks](#embedded-networks)
+-   [Style guide](#style-guide)
+    -   [The `@HapiTest` checklist](#the-hapitest-checklist)
+-   [JUnit Jupiter integrations](#junit-jupiter-integrations)
+    -   [Meta-annotations](#meta-annotations)
+    -   [`SharedNetworkLauncherSessionListener`](#sharednetworklaunchersessionlistener)
+    -   [`NetworkTargetingExtension`](#networktargetingextension)
+    -   [`SpecNamingExtension`](#specnamingextension)
+    -   [`SpecManagerExtension`](#specmanagerextension)
+    -   [`SpecEntityExtension`](#specentityextension)
 
 ## Structure of a `HapiSpec`
 
-A `HapiSpec` groups a sequence of `SpecOperation`s and gives them shared infrastructure and context. When
-the spec is executed against its target `HederaNetwork`, it runs each operation in sequence, and terminates
-with an exception if any operation fails. This is consistent with JUnit test semantics, where a test passes
-as long as no exception is thrown.
+A `HapiSpec` groups a sequence of `SpecOperation`s and gives them shared infrastructure and context.
+When the spec is executed against its target `HederaNetwork`, it runs each operation in sequence,
+and terminates with an exception if any operation fails. This is consistent with JUnit test
+semantics, where a test passes as long as no exception is thrown.
 
-Most operations are subclasses of `HapiSpecOperation`, which provides implementation support for submitting
-transactions and sending queries to the target network. The below diagram is a schematic of a `HapiSpec`.
+Most operations are subclasses of `HapiSpecOperation`, which provides implementation support for
+submitting transactions and sending queries to the target network. The below diagram is a schematic
+of a `HapiSpec`.
 
 <img src="./docs/assets/hapispec-schematic.png" width="900">
```

Comment:
Do we need a set width here?

If not I'd convert the image to markdown syntax.